### ISSUE TITLE
EU Covid Certificate not working with Spain Travel Health Formular

### ISSUE THREAD

**Issue by FlightMS** [corona-warn-app/cwa-documentation#778]
## Avoid duplicates
* [x] Bug is not mentioned in the [FAQ](https://www.coronawarn.app/en/faq/)
* [x] Bug affects both Android **and** iOS, for specific issues / questions that apply only to one operating system, please raise them in the respective repositories:
    * [Android repository](https://github.com/corona-warn-app/cwa-app-android)
    * [iOS repository](https://github.com/corona-warn-app/cwa-app-ios)
* [x] Bug is not already reported in another issue

## Technical details

- Device name: iPhone 11
- iOS/Android version: 15.2
- App version: 2.14.1

## Describe the bug

For entering Spain, you need to fill out the form "Spain Travel Health" formular (https://www.spth.gob.es/). One step is to upload a valid EU Covid Certificate. I've created a PDF from CWA and tried to validate but validation failed. So I generated another EU Covid Certificate PDF for the same vaccination within the CovPass app and here it worked without any problems.

Seems there is a slighty differece between both generation methods which will cause failing the validation when using the form.

## Steps to reproduce the issue

1. Open app
2. Generate EU Covid Certificate
3. Upload to Spain Travel Health formular
4. Click on "Validate"

## Expected behaviour

EU Covid Certificate will get validated without any problems

## Additional context

---
Internal Tracking ID: [EXPOSUREAPP-11074](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-11074)

**Comment by dsarkar** [comment]
HI @FlightMS Thanks for the report.
You tried both, iOS and Android CWA apps, and it does not work, correct?
You tried both, iOS and Android CovPass apps, and it does work, correct?

**Comment by dsarkar** [comment]
Internal Tracking ID: [EXPOSUREAPP-11074](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-11074)

---
Corona-Warn-App Open Source Team

**Comment by FlightMS** [comment]
Hi @dsarkar

I'm just able to try it with iOS devices but I presume it could be a general issue.
Maybe someone with Android can also verify it.

**Comment by dsarkar** [comment]
@FlightMS ok, thanks for the feedback.

**Comment by dsarkar** [comment]
@FlightMS BTW, they also have an app (SpTH), which allows to scan the DCC.

**Comment by FlightMS** [comment]
> @FlightMS BTW, they also have an app (SpTH), which allows to scan the DCC.

@dsarkar oh okay, thank you for the information. I've used the website on my device.

**Comment by moabits** [comment]
I just exported the certificate from both apps (CovPass + CWA) and on first sight I only notice one difference.
The CovPass app adds some additional margin to the exported PDF. The QR code content of both PDFs is identical.

**Comment by dsarkar** [comment]
@moabits thanks for testing, I forward this to the devs.

**Comment by dsarkar** [comment]
@moabits Android or iOS CWA?

**Comment by marcussc** [comment]
> FYI PR [corona-warn-app/cwa-app-ios#4041](https://github.com/corona-warn-app/cwa-app-ios/pull/4041)

As stated in the PR this is not the fix!

**Comment by MikeMcC399** [comment]
@FlightMS and others affected

Did you report the issue about the parsing of your files containing EU Digital COVID Certificates on the website https://www.spth.gob.es/ also to the contact mentioned on the website itself?

On the [EU COVID Digital Certificate validator](https://www.spth.gob.es/validator) form there is a link on the bottom right [Complaints and suggestions](mailto:oiac@mscbs.es) / [Beschwerden und Anregungen](mailto:oiac@mscbs.es).

**Comment by Ein-Tim** [comment]
@FlightMS Can you recheck if this issue still exists?

**Comment by FlightMS** [comment]
Still not working with a recently created DCC

![5A2B9CC7-C7B3-4C91-9FBE-43803AA3FD21](https://user-images.githubusercontent.com/82420268/163882661-cfe69b7f-2efd-46e8-b726-1ff111fe0272.png)

**Comment by MikeMcC399** [comment]
@FlightMS 

I tried on an Android phone with Google Chrome selecting the file `health_certificate_FirstName_LastName.pdf` and I got the message "No QR could be detected in the attached document ...."

I then tried zooming in to the QR code with a PDF viewer on the mobile device and taking a screenshot, which generated a jpg file. That was also not accepted. Same error message.

Then I printed out the certificate on DIN A4 paper size and selected this time the camera as input for the website https://www.spth.gob.es/validator. I then pointed the camera at the QR code from the print-out of `health_certificate_FirstName_LastName.pdf` and this was successfully validated.

It seems that the QR code detection of the website https://www.spth.gob.es/validator is very sensitive. It does not look like there was anything wrong with the file generated by CWA (although I did test on Android, not on iOS). If the QR code was incorrect then it would not have been accepted in the final step using the camera as input device.
